### PR TITLE
Wrote some comments to explain the A2C and PPO2 implementation

### PR DESCRIPTION
**Pull request by simoninithomas**
Hello,

1. I've added **some comments in A2C implementations** to help the users to better understand this implementation. I've tried to be succinct.
--> I think it can help readers who try to modify the implementations.

2. Also, I've **modified the readme file** to explain rapidly what each file do.

Have a great day,

### CODE REVIEW

**Comment by danzel** [comment]
Needs rebasing/redoing as the a2c/ppo2 implementations got refactored a bit recently.
Documentation for this would be very useful imho

**Comment by danzel** [comment]
`MlpPolicy` in the `policies.py` files could do with some comments for easier understanding too. (following the ones added to `CnnPolicy`)

File: baselines/ppo2/ppo2.py [hunk 1]
```diff
@@ -10,47 +10,113 @@
...
+        # Unclipped value
         vf_losses1 = tf.square(vpred - R)
+        # Clipped value
         vf_losses2 = tf.square(vpredclipped - R)
+        # Value loss 0.5 * SUM [max(unclipped, clipped)
         vf_loss = .5 * tf.reduce_mean(tf.maximum(vf_losses1, vf_losses2))
+        
+
+        # Remember we want ratio (pi current policy / pi old policy)
+        # But neglopac returns us -log(policy)
```
Comment (danzel) [inline baselines/ppo2/ppo2.py]: `neglopac` -> `neglogpac`

File: baselines/ppo2/ppo2.py [hunk 2]
```diff
@@ -86,23 +152,40 @@ def load(load_path):
...
 
     def run(self):
+        # Here, we init the lists that will contain the mb of experiences
         mb_obs, mb_rewards, mb_actions, mb_values, mb_dones, mb_neglogpacs = [],[],[],[],[],[]
         mb_states = self.states
         epinfos = []
+
+        # For n in range number of steps
         for _ in range(self.nsteps):
+            # Given observations, get action value and neglopacs
```
Comment (danzel) [inline baselines/ppo2/ppo2.py]: `neglopacs` -> `neglogpacs`

File: baselines/ppo2/ppo2.py (hunk 1 above)
Comment (simoninithomas) [reply baselines/ppo2/ppo2.py]: Thanks for the feedback I will correct the typos. Quick question do you think I send a new pull request or I just update these ?

Comment (danzel) [reply baselines/ppo2/ppo2.py]: I've never dealt with this repo before, but usually I'd just update the existing PR.

**Comment by araffin** [comment]
@simoninithomas I don't know what is the current status of this PR. However, I would be interested in integrating your comments into the stable-baselines repo (a fork of OpenAI Baselines): https://github.com/hill-a/stable-baselines/
The only issue is that if I integrate your comments myself, no trace of your authorship will be kept.

**Comment by simoninithomas** [comment]
@araffin Hi, I've just read your article that explains this project and that's amazing. As you want I can make a pull request on your fork or you can add them directly.

**Comment by araffin** [comment]
@simoninithomas I would prefer a PR ;) (we are quite busy right now with conference deadlines coming...)

**Comment by pzhokhov** [comment]
This looks great, but, as @danzel mentioned, we have refactored the code quite a bit. @simoninithomas if still interested in merging, we can work together to resolve conflicts.

**Comment by simoninithomas** [comment]
Hi @pzhokhov I'm interested in merging, do you prefer that I fork the new baseline version, add the comments and make a new pull request ?

**Comment by pzhokhov** [comment]
that or update this PR - either works

**Comment by pzhokhov** [comment]
also, feel free to ask questions here about the new version if something is not clear / can be clarified.

**Comment by simoninithomas** [comment]
I deleted my former fork and re-fork baselines (it was easier to do this way because there was too much conflicting files.
Consequently I need to make a new PR, so I close this one and open a new one.

**Pull request closed without merging**